components/nvs_flash/src/nvs_storage.cpp

```diff
@@ -269,6 +269,14 @@ esp_err_t Storage::writeItem(uint8_t nsIndex, ItemType datatype, const char* key
         VerOffset prevStart,  nextStart;
         prevStart = nextStart = VerOffset::VER_0_OFFSET;
         if (findPage) {
+            // Do a sanity check that the item in question is actually being modified.
+            // If it isn't, it is cheaper to purposefully not write out new data.
+            // since it may invoke an erasure of flash.
+            if (ESP_ERR_NVS_COMPARE_IDENTICAL == cmpMultiPageBlob(nsIndex, key, data, dataSize))
```

Comment:
Here and below, can you please write the conditions in "normal" order (`something_complex == constant`)? The compiler is capable of detecting accidental assignments in conditional expressions, so no need to reverse them.